Zend/zend_language_scanner.h

```diff
@@ -63,6 +64,12 @@ typedef struct _zend_heredoc_label {
 	zend_bool indentation_uses_spaces;
 } zend_heredoc_label;
 
+/* Track locations of unclosed {, [, (, etc. for better syntax error reporting */
+typedef struct _zend_nest_location {
+	char *text; /* points into yytext */
```

Comment:
Do we actually need to store the pointer rather than just `char opening`?

Reply:
I tried storing just a `char` and benchmarking, but found it wasn't any faster. The original reason for using `char*` was in case there are any other nested syntactic constructs which can't be disambiguated from a single character... but looking at the PHP grammar, I haven't seen any such constructs yet.

I'll change it to `char`.